app/connectors/util/CustomHttpReader.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.util
+
+import play.api.Logger
+import play.api.http.Status
+import uk.gov.hmrc.http.{HttpErrorFunctions, HttpReads, HttpResponse}
+
+object CustomHttpReader extends HttpReads[HttpResponse] with HttpErrorFunctions with Status {
+  override def read(method: String, url: String, response: HttpResponse): HttpResponse = {
+    Logger.debug(s"CustomHttpReader Log\nstatus: ${response.status}\nbody: ${response.body}\nheaders: ${response.headers.map {
+      x =>
+        s"\n  ${x._1} : ${x._2}"
+    }}")
```

Comment:
Maybe tuple would be more informative in that case? I mean sth like:
`(firstValue, secondValue) => s"\n  ${firstValue} : ${secondValue}"`

Reply:
Tried that, and it doesn't work. It doesn't really change anything though.

Reply:
What about readability so in the future we know what `x._1` and `x._2` are?